src/leappto/cli.py

```diff
@@ -676,42 +519,42 @@ def print_migrate_info(text):
         src_ip = machine_src.ip[0]
         dst_ip = machine_dst.ip[0]
 
-        user_mapped_ports = PortMap()
-        user_excluded_ports = PortList()
-
-        tcp_mapping = None
-        #udp_mapping = None
-
-        try:
-            if parsed.forwarded_tcp_ports:
-                for target_port, source_port in parsed.forwarded_tcp_ports:
-                    user_mapped_ports.set_tcp_port(source_port, target_port)
-
-            if parsed.excluded_tcp_ports:
-                for target_port, source_port in parsed.excluded_tcp_ports:
-                    user_excluded_ports.set_tcp_port(source_port)
-
-
-            if not parsed.ignore_default_port_map:
-                print_migrate_info('! Scanning source ports')
-                src_ports = _port_scan(src_ip, shallow=True)
-            else:
-                src_ports = PortList()
-
-            print_migrate_info('! Scanning target ports')
-            dst_ports = _port_scan(dst_ip, shallow=True)
-
-            tcp_mapping = _port_remap(src_ports, dst_ports, user_mapped_ports, user_excluded_ports)["tcp"]
+        print_migrate_info('! configuring SSH keys')
 
-        except PortCollisionException as e:
-            print(str(e))
-            exit(-1)
-        except PortScanException as e:
-            print("An error occured during port scan: {}".format(str(e)))
-            exit(-1)
+        mc = MigrationContext(
+            machine_dst,
+            _set_ssh_config(parsed.target_user, parsed.target_identity, parsed.target_ask_pass),
+            None if parsed.use_rsync else machine_src.disks[0].host_path,
+            machine_src,
+            _set_ssh_config(parsed.source_user, parsed.source_identity, parsed.source_ask_pass),
+            parsed.use_rsync
+        )
 
+        if not parsed.print_port_map:
+            # If we're doing an actual migration, check we have access to the
+            # target, and the desired container name is available
+            check_result, claimed_names = mc.check_target()
+            if check_result != 0:
+                print("! Checking target access failed")
+                sys.exit(check_result)
+
+            container_name = mc.get_target_container_name()
+            if container_name in claimed_names:
+                print("! Container name {} is not available".format(container_name))
+                sys.exit(-10)
```

Comment:
I wonder if we should start using errno's as exit codes (plenty of tools do) and rather return `errno.EEXIST` here since `-10` feels rather arbitrary.